sdk/src/azure/core/az_json_writer.c

```diff
@@ -799,7 +801,7 @@ az_json_writer_append_json_text(az_json_writer* ref_json_writer, az_span json_te
   // Therefore, need_comma must be true after appending the json_text.
 
   // We already tracked and updated bytes_written while writing, so no need to update it here.
-  _az_update_json_writer_state(ref_json_writer, 0, az_span_size(json_text), true, last_token_kind);
+  _az_update_json_writer_state(ref_json_writer, 0, required_size, true, last_token_kind);
```

Comment:
Though this fixes up `_internal.total_bytes_written`, this seems to be incrementing `_internal.bytes_written` by one more than necessary, because it is already being incremented on line 789.

Reply:
Nevermind, I see the bytes_written increment parameter continues to be 0.